src/Umbraco.Core/Services/Implement/ContentService.cs

```diff
@@ -757,6 +757,11 @@ public OperationResult Save(IContent content, int userId = Constants.Security.Su
             if (publishedState != PublishedState.Published && publishedState != PublishedState.Unpublished)
                 throw new InvalidOperationException("Cannot save (un)publishing content, use the dedicated SavePublished method.");
 
+            if (content.Name.Length > 255)
+            {
+                throw new Exception("Name cannot be more than 255 characters in length.");
```

Comment:
This should be `InvalidOperationException`. Throwing the base class `Exception` is not best practices since you cannot then catch an explicit exception type.